kindelia_core/src/runtime/mod.rs

```diff
@@ -1631,7 +1678,28 @@ impl Runtime {
             let cont = ask_arg(self, term, 1);
             let normalized = self.normalize(unnormalized, mana, &mut HashSet::new())?;
             let cont = alloc_app(self, cont, normalized);
-            let done = self.run_io(subject, caller, cont, mana);
+            let done = self.run_io(subject, caller, cont, mana, call_term, events);
+            clear(self, host, 1);
+            clear(self, term.get_loc(0), 2);
+            return done;
+          }
+          IO_EMIT => {
+            let term_to_emit = ask_arg(self, term, 0);
+            // checks if args is a number
+            if self.check_num(term_to_emit, mana).is_err() {
+              // if not a number, checks if is a flat CTR
+              self.check_ctr_argm(subject, None, mana, term_to_emit)?;
```

Comment:
I'm not sure we should limit this the same way as `CALL`. I'll review this better tomorrow.

Reply:
In the  issue #183 is written:
> Emitted terms would be limited somehow, so it's O(1) to read them. Could be in the same way contract call input is limited: only allowing flat Ctrs on U120s.

The way is done, is only allowing "flat" `Ctrs` or pure `U120`